Test failure due to mypy update

```
/home/ys/Projects/python/result/tests/type-checking/test_result.yml:51:
E   pytest_mypy_plugins.utils.TypecheckAssertionError: Invalid output:
E   Actual:
E     ../../home/ys/Projects/python/result/.venv/lib/python3.9/site-packages/mypy/typeshed/stdlib/email/message.pyi:13: error: Cannot determine type of "Any" (diff)
E     ../../home/ys/Projects/python/result/src/result/result:39: note: __match_args__ must be final for checking of match statements to work (diff)
E     ../../home/ys/Projects/python/result/src/result/result:176: note: __match_args__ must be final for checking of match statements to work (diff)
E     main:5: error: Incompatible types in assignment (expression has type "Ok[float]", variable has type "Ok[int]") (diff)
E     main:9: error: Incompatible types in assignment (expression has type "Err[Exception]", variable has type "Err[TypeError]") (diff)
E     main:13: error: Incompatible types in assignment (expression has type "Union[Ok[float], Err[Exception]]", variable has type "Union[Ok[int], Err[TypeError]]") (diff)
E   Expected:
E     main:5: error: Incompatible types in assignment (expression has type "Ok[float]", variable has type "Ok[int]") (diff)
E     main:9: error: Incompatible types in assignment (expression has type "Err[Exception]", variable has type "Err[TypeError]") (diff)
E     main:13: error: Incompatible types in assignment (expression has type "Union[Ok[float], Err[Exception]]", variable has type "Union[Ok[int], Err[TypeError]]") (diff)
E   Alignment of first line difference:
E     E: main:5: error: Incompatible types in assignment (expression has type "Ok...
E     A: ../../home/ys/Projects/python/result/.venv/lib/python3.9/site-packages/m...
E        ^
```

Latest mypy seems to remove asterisk syntax: https://github.com/python/mypy/pull/12459. Currently tests are failed with mypy >= 9.40

--- a/tests/type-checking/test_result.yml
+++ b/tests/type-checking/test_result.yml
@@ -29,17 +29,17 @@
     res1 = Ok()
     reveal_type(res1) # N: Revealed type is "result.result.Ok[builtins.str]"
     res2 = Ok(42)
-    reveal_type(res2) # N: Revealed type is "result.result.Ok[builtins.int*]"
+    reveal_type(res2) # N: Revealed type is "result.result.Ok[builtins.int]"
     res3 = Err(1)
-    reveal_type(res3) # N: Revealed type is "result.result.Err[builtins.int*]"
+    reveal_type(res3) # N: Revealed type is "result.result.Err[builtins.int]"
 
     res4 = Ok(4)
     add1: Callable[[int], Result[int, str]] = lambda i: Ok(i + 1)
     toint: Callable[[str], Result[int, str]] = lambda i: Ok(int(i))
     res5 = res4.and_then(add1)
-    reveal_type(res5) # N: Revealed type is "Union[result.result.Ok[builtins.int*], result.result.Err[builtins.str*]]"
+    reveal_type(res5) # N: Revealed type is "Union[result.result.Ok[builtins.int], result.result.Err[builtins.str]]"
     res6 = res4.or_else(toint)
-    reveal_type(res6) # N: Revealed type is "Union[result.result.Ok[builtins.int], result.result.Err[builtins.str*]]"
+    reveal_type(res6) # N: Revealed type is "Union[result.result.Ok[builtins.int], result.result.Err[builtins.str]]"
 
 - case: covariance
   disable_cache: false

